src/main/java/com/google/edith/servlets/ReceiptData.java

```diff
@@ -0,0 +1,95 @@
+package com.google.edith.servlets;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.cloud.documentai.v1beta2.DocumentUnderstandingServiceClient;
+import com.google.edith.interfaces.ExtractReceiptInterface;
+import com.google.edith.services.ExtractReceiptService;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/** Converts parsed receipt file string to receipt and item objects. */
+public final class ReceiptData {
+
+  private ExtractReceiptInterface extractReceiptImplementation;
+
+  public ReceiptData() throws IOException {
+    this.extractReceiptImplementation =
+        new ExtractReceiptService(DocumentUnderstandingServiceClient.create());
+  }
+
+  public ReceiptData(ExtractReceiptInterface extractReceiptImplementation) {
+    this.extractReceiptImplementation = extractReceiptImplementation;
+  }
+
+  /**
+   * Creates receipt object.
+   *
+   * @param blobKey - string representation of the blob in GCS bucket.
+   * @param expenditureName - name of the rceipt obtained from FE form.
+   * @return Receipt - receipt object created from the provided info.
+   */
+  public Receipt extractReceiptData(String blobKey, String expenditureName) throws IOException {
+
+    UserService userService = UserServiceFactory.getUserService();
+    User user = userService.getCurrentUser();
+
+    List<Map<String, String>> items = extractReceiptImplementation.extractReceipt(blobKey);
+    Item[] parsedItems = createReceiptItems(user, items);
+    Receipt parsedReceipt = createReceipt(blobKey, user, parsedItems, expenditureName);
+    return parsedReceipt;
+  }
+
+  /**
+   * Creates receipt object.
+   *
+   * @param blobKey - string representation of the blob in GCS bucket.
+   * @param user - user who is logged in.
+   * @param items - Array of items parsed from receipt pdf file.
+   * @param expenditureName - name of the rceipt obtained from FE form.
+   * @return Receipt - receipt object created from the provided info.
+   */
+  private Receipt createReceipt(String blobKey, User user, Item[] items, String expenditureName) {
+    Receipt userReceipt =
+        new Receipt(
+            user.getUserId(),
+            "unknown store name",
+            "unknown date",
+            expenditureName,
+            blobKey,
+            0.0f,
+            items);
+    return userReceipt;
+  }
+
+  /**
+   * Creates Array of Item objects from list of item description .
+   *
+   * @param user - user who is logged in.
+   * @param extractedData - List of item description.
```

Comment:
What is the map inside this list expected to contain? What are the key value pairs?

Reply:
also someone else's file, just an unmerged dependency